reactor-core/src/main/java/reactor/core/publisher/UnicastManySinkNoBackpressure.java

```diff
@@ -182,18 +216,31 @@ public void emitComplete() {
 	public Emission tryEmitComplete() {
 		switch (state) {
 			case INITIAL:
-				// TODO different Emission?
-				return Emission.OK;
+				if (STATE.compareAndSet(this, State.INITIAL, State.TERMINATED_BEFORE_SUBSCRIPTION)) {
+					return Emission.OK;
+				}
+				else {
+					//recurse
```

Comment:
WDYT about changing it to `for (...)` to avoid the recursion (== longer stacktrace)?

Reply:
done